src/linpred.jl

```diff
@@ -126,7 +131,7 @@ function cholesky(p::DensePredChol{T}) where T<:FP
 end
 cholesky!(p::DensePredQR{T}) where {T<:FP} = Cholesky{T,typeof(p.X)}(p.qr.R, 'U', 0)
 
-function delbeta!(p::DensePredChol{T,<:Cholesky}, r::Vector{T}) where T<:BlasReal
+function delbeta!(p::DensePredChol{T,<:Cholesky}, r::Vector{T}; allowrankdeficient::Bool=false) where T<:BlasReal
```

Comment:
Is the argument used in the function at all?

Reply:
It is not. But if I don't include it in the signature, then you get an error when using `allowrankdeficient=false`, and the tests will fail.

Reply:
It probably just needs to dispatch on the correct `Type{p}`... it should be fine for now, but this seems like another pro-having it as a field to simplify the code.